addons/mail/static/src/discuss/core/public_web/discuss_sidebar_categories.xml

```diff
@@ -104,7 +104,7 @@
                 <t t-set="isCategoryOpen" t-value="thread.discussAppCategory.open" />
                 <ul t-if="subChannels.length > 0" class="list-unstyled position-relative flex-grow-1" t-att-class="{ 'my-1 d-flex flex-column gap-1': store.discuss.isSidebarCompact, 'my-0 d-flex flex-column gap-1': !store.discuss.isSidebarCompact }">
                     <t t-foreach="subChannels" t-as="sub" t-key="sub.localId">
-                        <DiscussSidebarSubchannel t-if="isCategoryOpen or sub.eq(store.discuss.thread)" thread="sub" isFirst="sub_first or !isCategoryOpen"/>
+                        <DiscussSidebarSubchannel t-if="sub.eq(store.discuss.thread) or (isCategoryOpen and (!thread.isMuted or sub.selfMember?.message_unread_counter > 0))" thread="sub" isFirst="sub_first or !isCategoryOpen"/>
```

Comment:
When parent channel is selected, pinned threads must be shown, as long as they are not muted.
Pinned threads are not shown on unselected parent channel, unless they have unread messages.
`isCategoryOpen` should be for the case neither thread nor the parent channel is selected.